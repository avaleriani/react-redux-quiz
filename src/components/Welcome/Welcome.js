import React from "react";
import styles from "./styles.module.css";

/**
 * @param {{startGame: function}} props
 * @return HTMLElement
 */
const Welcome = (props) => (
  <div>
    Start quiz
    <button onClick={props.startGame}>Start</button>
  </div>
);

export default Welcome;