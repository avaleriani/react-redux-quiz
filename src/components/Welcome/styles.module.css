.welcome {
    display: flex;
    flex-direction: column;
}

.welcome img {
    width: 256px;
    height: 256px;
    transform-origin: 50% 50%;
    opacity: 0;
    padding-bottom: 50px;
    animation: grow 700ms forwards;
}

@keyframes grow {
    0% {
        opacity: 0;
        transform: scale(.0001) rotateZ(-360deg);
    }
    100% {
        opacity: 1;
        transform: scale(.99) rotateZ(360deg);
    }
}