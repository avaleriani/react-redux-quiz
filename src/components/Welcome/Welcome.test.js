import React from 'react';
import renderer from 'react-test-renderer';
import Welcome from './Welcome';

describe('Welcome component', () => {

  test('renders correctly without props', () => {
    const WelcomeComponent = renderer.create(<Welcome/>).toJSON();
    expect(WelcomeComponent).toMatchSnapshot();
  });

  test('renders with props', () => {
    const props = {
      startGame: () => {
      }
    };
    const WelcomeComponent = renderer.create(<Welcome {...props}/>).toJSON();
    expect(WelcomeComponent).toMatchSnapshot();
  });
});