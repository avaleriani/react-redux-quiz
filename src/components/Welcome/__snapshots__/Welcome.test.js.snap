// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component renders correctly without props 1`] = `
<div
  className="welcome"
>
  <img
    alt="logo"
    src="/images/logo.png"
  />
  <button
    className="button"
    onClick={null}
    type="button"
  >
    Start
  </button>
</div>
`;

exports[`Welcome component renders with props 1`] = `
<div
  className="welcome"
>
  <img
    alt="logo"
    src="/images/logo.png"
  />
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Start
  </button>
</div>
`;
